cpu/stm32_common/periph/i2c_1.c

```diff
@@ -451,17 +480,53 @@ static inline int _stop(I2C_TypeDef *i2c)
     DEBUG("[i2c] stop: Wait for transfer to be complete\n");
     while (!(i2c->ISR & I2C_ISR_TC) && tick--) {
         if (i2c->ISR & ERROR_FLAG || !tick) {
-            return -ETIMEDOUT;
+            return -EIO;
         }
     }
 
     /* send STOP condition */
     DEBUG("[i2c] stop: Generate stop condition\n");
     i2c->CR2 |= I2C_CR2_STOP;
 
+    /* Wait for the stop to complete */
+    tick = TICK_TIMEOUT;
+    while (!(i2c->CR2 & I2C_CR2_STOP) && tick--) {
+        if (!tick) {
```

Comment:
move `if` after the while

Reply:
I agree and I addressed this kind of comments in the last fixup commit of this PR